tools/build-tools/src/test/typeValidator/classValidator.spec.ts

```diff
@@ -49,92 +49,503 @@ describe("Class", () => {
+            const classOld =
+            `
+            class BaseClass<T> extends Promise<T> {}
+            export class TestClass extends BaseClass<string> {}
+            `;
+
+            const classNew =
+            `
+            class BaseClass2<T> extends Promise<T> {}
+            export class TestClass extends BaseClass2<string> {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // changing implementations is breaking regardless of structural identicality
+        // (this isn't actually breaking but is difficult to handle and errs on the side
+        // of over-bumping)
+        it("changed class implementations that are structurally identical", () => {
+            const classOld =
+            `
+            interface TestInterface {}
+            export class TestClass implements TestInterface {}
+            `;
+
+            const classNew =
+            `
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface2 {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // reordering implementations is not breaking
+        it("reorders changed class implementations", () => {
+            const classOld =
+            `
+            interface TestInterface1 {}
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface1, TestInterface2 {}
+            `;
+
+            const classNew =
+            `
+            interface TestInterface1 {}
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface2, TestInterface1 {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.none);
+        });
+    });
+
+    describe("type parameters", () => {
+        // adding/removing class type parameters is not a breaking change if they
+        // aren't also exposed on the public API elsewhere
+        it("adds a class type parameters not exposed on the public API", () => {
+            const classOld =
+            `
+            export class TestClass {
+                private prop1?: Array<string>;
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass<X> {
+                private prop1?: Array<X>;
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.none);
+        });
+
+        // class type parameter changes exposed on the public API are the same as
+        // equivalent typing changes
+        it.skip("changes a class type parameters exposed on the public API", () => {
```

Comment:
Do you mean to skip this?

Reply:
addressed as part of addressing the other comments on the type parameter tests.  type parameter handling isn't fully working atm